pkg/providers/vsphere/vcenter/getvm.go

```diff
@@ -25,18 +25,28 @@ func GetVirtualMachine(
 	datacenter *object.Datacenter,
 	finder *find.Finder) (*object.VirtualMachine, error) {
 
-	if instanceUUID := vmCtx.VM.UID; instanceUUID != "" {
-		if vm, err := findVMByUUID(vmCtx, vimClient, datacenter, string(instanceUUID), true); err == nil {
+	// Find by instance UUID.
+	if id := vmCtx.VM.UID; id != "" {
+		if vm, err := findVMByUUID(vmCtx, vimClient, datacenter, string(id), true); err == nil {
 			return vm, nil
 		}
 	}
 
-	if uniqueID := vmCtx.VM.Status.UniqueID; uniqueID != "" {
-		if vm, err := findVMByMoID(vmCtx, finder, uniqueID); err == nil {
+	// Find by BIOS UUID.
```

Comment:
Do we need a comment for @dougm to remove this and use the spec.InstanceUUID here?

Reply:
I do not think so. Having a fallback to BIOS UUID is not going to hurt us, and is preferable to falling all the way back to the inventory path. vC returns an error if there are multiple results, so this is fine I think.